pkg/extensions/sync/sync.go

```diff
@@ -349,14 +386,15 @@ func syncRegistry(regCfg RegistryConfig, upstreamURL string, storeController sto
 	}
 
 	for _, ref := range images {
-		upstreamImageRef := ref
+		upstreamImageRef := ref.ref
 
-		repo := getRepoFromRef(upstreamImageRef, upstreamAddr)
+		remoteRepo := getRepoFromRef(upstreamImageRef, upstreamAddr)
+		localRepo := getRepoDestination(remoteRepo, ref.destination)
 		tag := getTagFromRef(upstreamImageRef, log).Tag()
 
-		imageStore := storeController.GetImageStore(repo)
+		imageStore := storeController.GetImageStore(remoteRepo)
```

Comment:
this should be localRepo instead of remoteRepo, right? imageStore is a local image store.

We want to compare both upstream and downstream images